src/python/pants/backend/jvm/tasks/jvm_compile/compile_context.py

```diff
@@ -8,18 +8,20 @@
 import zipfile
 from contextlib import contextmanager
 
+from pants.backend.jvm.targets.jar_library import JarLibrary
 from pants.build_graph.aliased_target import AliasTarget
 from pants.build_graph.target import Target
 from pants.util.contextutil import open_zip
 
 
 def _resolve_strict_dependencies(target):
```

Comment:
I think the question I have here is whether all dependencies of JarLibrary should be exported by default. I suppose a jar_library target has dependencies only because the compilation will fail without it, even in the pre-strict_deps era. Is there use cases where NOT all dependencies of a jar_library are needed for its dependees to compile?